GW2EIEvtcParser/EncounterLogic/Raids/W8/DecimaTheStormsinger.cs

```diff
@@ -121,35 +170,274 @@ internal override void ComputeNPCCombatReplayActors(NPC target, ParsedEvtcLog lo
+                    {
+                        foreach (EffectEvent effect in foreshockLeft.Where(x => x.Time >= start && x.Time < end))
+                        {
+                            (long start, long end) lifespanLeft = effect.ComputeLifespan(log, 1967);
+                            nextStartTime = lifespanLeft.end;
+                            var rotation = new AngleConnector(effect.Rotation.Z + 90);
+                            var leftHalf = (PieDecoration)new PieDecoration(1185, 180, lifespanLeft, Colors.LightOrange, 0.4, new PositionConnector(effect.Position)).UsingRotationConnector(rotation);
+                            replay.AddDecorationWithBorder(leftHalf, Colors.LightOrange, 0.6);
+                        }
+                    }
+
+                    // Decima's Right Side
+                    if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.DecimaForeshockRight, out var foreshockRight))
+                    {
+                        foreach (EffectEvent effect in foreshockRight.Where(x => x.Time >= start && x.Time < end))
+                        {
+                            (long start, long end) lifespanRight = effect.ComputeLifespan(log, 3000);
+                            lifespanRight.start = nextStartTime - 700; // Trying to match in game timings
+                            nextStartTime = lifespanRight.end;
+                            var rotation = new AngleConnector(effect.Rotation.Z + 90);
+                            var rightHalf = (PieDecoration)new PieDecoration(1185, 180, lifespanRight, Colors.LightOrange, 0.4, new PositionConnector(effect.Position)).UsingRotationConnector(rotation);
+                            replay.AddDecorationWithBorder(rightHalf, Colors.LightOrange, 0.6);
+                        }
+                    }
+
+                    // Decima's Frontal
+                    if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.DecimaForeshockFrontal, out var foreshockFrontal))
+                    {
+                        foreach (EffectEvent effect in foreshockFrontal.Where(x => x.Time >= start && x.Time < end))
+                        {
+                            (long start, long end) lifespanFrontal = effect.ComputeLifespan(log, 5100);
+                            lifespanFrontal.start = nextStartTime;
+                            var frontalCircle = new CircleDecoration(600, lifespanFrontal, Colors.LightOrange, 0.4, new PositionConnector(effect.Position));
+                            replay.AddDecorationWithBorder(frontalCircle, Colors.LightOrange, 0.6);
+                        }
+                    }
+                }
+
+                // Earthrend - Outer Sliced Doughnut - 8 Slices
+                if (log.CombatData.TryGetEffectEventsBySrcWithGUID(target.AgentItem, EffectGUIDs.DecimaEarthrendDoughnutSlice, out var earthrend))
```

Comment:
TryGetGroupedEffectEventsBySrcWithGUID then use the first element of each subarray